### DISCUSSION TITLE
Support defining Kubernetes resources without Kubernetes API data types and with list and dicts

### DISCUSSION THREAD

**Discussion by ketozhang** [apache/airflow#37123] [Ideas]
### Description

`KubernetesPodOperator` and operator's `pod_override` support the K8s Python API data models/types and in some cases also supports passing in Python standard data types (e.g., `env_vars`, `cmds`, etc).

It would be nice to expand the cases to allow defining all K8s spec with Python standard data types instead of using the API's data types. Users would have to sacrifice type integrity, but they have options to use type hinting (e.g., with the K8s type stubs package).

Current alternative is to rely on `pod_template_dict`.

### Use case/motivation

- [ ] `pod_override: dict | V1Pod`
- [ ] `ports: list[int] | list[V1ContainerPort]`
- [ ] `volume_mounts: list[dict] | list[V1VolumeMount]`
- [ ] `volumes: list[dict] | list[V1Volume]`
- and other KPO fields ...

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by ketozhang** [comment]
I'd like to contribute to this, but I am unaware if this was attempted before and perhaps some limitation in the K8s API prevents this. Naively, I don't understand what possible implementation detail prevents us from doing:

One option is to rely on a private method as it's how `pod_template_dict` is implemented, but beyond deserializing V1Pod,

https://github.com/apache/airflow/blob/ba053f79edab54366208aad9d88877a95b789eca/airflow/providers/cncf/kubernetes/pod_generator.py#L559-L572

There are few others options involving writing our own deserializer or just serialize to YAML and pass it similar to `pod_template_file`.

**Comment by dirrao** [comment]
@ketozhang
Right now the pod operator is overloaded with too many details. It would be great if you can simplify it.

**Comment by dimberman** [comment]
@ketozhang can you make a DAG example showing how this would look? We had something kind of like this in earlier versions of the k8sexecutor and it did not go very well. I'm open to hearing more but want to be careful.

**Comment by cmarteepants** [comment]
This sounds similar to how KPOs and k8sexecutor configs were defined pre-2.0. See https://github.com/apache/airflow/pull/10393 for context for this was removed. Simplifying would be amazing, but agree with @dimberman that we need to tread carefully. I am going to convert this into a discussion so we can brainstorm examples of how this will look.

**Reply by ketozhang** [reply]
Taken a look at #10393. I couldn't find much motivation. @dimberman mentioned some previous email thread. I know it's been many years; would you happen to have a link to this or perhaps a quick summary. 

PS: I haven't forgotten about making an example. TBD...
